pytensor/tensor/rewriting/linalg.py

```diff
@@ -407,23 +407,23 @@ def rewrite_det_diag_from_eye_mul(fgraph, node):
     list of Variable, optional
         List of optimized variables, or None if no optimization was performed
     """
-    node_input_op = node.inputs[0]
+    node_input = node.inputs[0]
     # Check if the op is Elemwise and mul
     if not (
-        node_input_op.owner is not None
-        and isinstance(node_input_op.owner.op, Elemwise)
-        and isinstance(node_input_op.owner.op.scalar_op, Mul)
+        node_input.owner is not None
+        and isinstance(node_input.owner.op, Elemwise)
+        and isinstance(node_input.owner.op.scalar_op, Mul)
     ):
         return None
 
     # Find whether any of the inputs to mul is Eye
-    inputs_to_mul = node_input_op.owner.inputs
+    inputs_to_mul = node_input.owner.inputs
     eye_input = [
         mul_input
         for mul_input in inputs_to_mul
         if mul_input.owner and isinstance(mul_input.owner.op, Eye)
     ]
-    if not (eye_input[0]):
+    if not (eye_input):
```

Comment:
I think you have to check that eye_input.broadcstable[:-2] is (False, False). Thinking about things like `pt.eye(1) * pt.ones((5, 5))` which I think this rewrite would return a wrong value.

Reply:
yea i checked the eye_input.broadcastable and its always (False, False). that logically does make sense because eye_input is always the identity matrix which is being multiplied w the scalar/vector/matrix

Reply:
What about eye(1) * full rank matrix?

Reply:
eye(1) holds no information about its shape or .broadcastable which makes it tough to address during the rewrite @jessegrabowski

Reply:
Eye is currently written in a way that discards shape information, I think we should fix this. It should be enough to just change [this line](https://github.com/pymc-devs/pytensor/blob/28d9d4dc0bd0fc0060670622514b83cc281cda3e/pytensor/tensor/basic.py#L1293) to set the shape of the returned tensor to (n,m) ?

Reply:
The alternative is to abort this rewrite if no shape information is available (i.e. if `eye_inputs[0].type.shape[-2:] == (None, None` or `non_eye_inputs[0].type.shape[-2:] == (None, None)`. But this will always be true because we can never have an Eye with shape the way it's implemented.

There are certain non-disruptive Ops, like `SpecifyShape`, that we could potentially look up one more level to check for an `Eye`, but it would add another layer of complexity here.

Reply:
We merged #841 last night that should unblock you @tanish1729. You will need to rebase from main, but then the `broadcastable[-2:] == (True, True)` check should catch the `eye(1)` case.

For the non-square case it's still a bit complicated. `x = eye(3, 2)` will have `x.type.shape = (3,2)` and we could just check that, but we couldn't check this:

```
i = pt.lscalar('i)
x = pt.eye(2, i)
x.type.shape == (2, None)
```

So we'll have to think a little bit about this last sticky case.